gst/nnstreamer/tensor_data.c

```diff
@@ -0,0 +1,296 @@
+      break;
+    case _NNS_FLOAT32:
+      td_get_data (td, value, float);
+      break;
+    case _NNS_INT64:
+      td_get_data (td, value, int64_t);
+      break;
+    case _NNS_UINT64:
+      td_get_data (td, value, uint64_t);
+      break;
+    default:
+      nns_logw ("Unknown tensor type %d", td->type);
+      return FALSE;
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief Typecast tensor element data.
+ * @param td struct for tensor data
+ * @param type tensor type to be transformed
+ * @return TRUE if no error
+ */
+gboolean
+gst_tensor_data_typecast (tensor_data_s * td, tensor_type type)
+{
+  gboolean is_float;
+
+  g_return_val_if_fail (td != NULL, FALSE);
+  g_return_val_if_fail (type != _NNS_END, FALSE);
+
+  /* do nothing when transform to same type */
+  if (td->type != type) {
+    is_float = (td->type == _NNS_FLOAT32 || td->type == _NNS_FLOAT64);
+
+    switch (type) {
+      case _NNS_INT32:
+        td_typecast (td, int32_t);
+        break;
+      case _NNS_UINT32:
+        if (is_float) {
+          td_typecast (td, int32_t);
+          td->type = _NNS_INT32;
+        }
+        td_typecast (td, uint32_t);
+        break;
+      case _NNS_INT16:
+        td_typecast (td, int16_t);
+        break;
+      case _NNS_UINT16:
+        if (is_float) {
+          td_typecast (td, int16_t);
+          td->type = _NNS_INT16;
+        }
+        td_typecast (td, uint16_t);
+        break;
+      case _NNS_INT8:
+        td_typecast (td, int8_t);
+        break;
+      case _NNS_UINT8:
+        if (is_float) {
+          td_typecast (td, int8_t);
+          td->type = _NNS_INT8;
+        }
+        td_typecast (td, uint8_t);
+        break;
+      case _NNS_FLOAT64:
+        td_typecast (td, double);
+        break;
+      case _NNS_FLOAT32:
+        td_typecast (td, float);
+        break;
+      case _NNS_INT64:
+        td_typecast (td, int64_t);
+        break;
+      case _NNS_UINT64:
+        if (is_float) {
+          td_typecast (td, int64_t);
+          td->type = _NNS_INT64;
+        }
+        td_typecast (td, uint64_t);
+        break;
+      default:
+        nns_logw ("Unknown tensor type %d", type);
+        return FALSE;
+    }
+
+    td->type = type;
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief Typecast tensor element value.
+ * @param input pointer of input tensor data
+ * @param in_type input tensor type
+ * @param output pointer of output tensor data
+ * @param out_type output tensor type
+ * @return TRUE if no error
+ */
+gboolean
+gst_tensor_data_raw_typecast (gpointer input, tensor_type in_type,
+    gpointer output, tensor_type out_type)
+{
+  tensor_data_s td;
+
+  g_return_val_if_fail (input != NULL, FALSE);
+  g_return_val_if_fail (output != NULL, FALSE);
+  g_return_val_if_fail (in_type != _NNS_END, FALSE);
+  g_return_val_if_fail (out_type != _NNS_END, FALSE);
+
+  gst_tensor_data_set (&td, in_type, input);
+  gst_tensor_data_typecast (&td, out_type);
+  gst_tensor_data_get (&td, output);
```

Comment:
return false if one of them fails?

Reply:
IMO, these lines are always true with the conditions in line 253~256

Reply:
I see. Thank you for the clarification :)